src/main/kotlin/com/procurify/flagcounter/FlagCounter.kt

```diff
@@ -90,13 +98,28 @@ class FlagCounter(
     private fun formatTeamMessage(teamFlags: List<FlagDetail>): String {
         // TODO Parameterize link url based on project/environment configuration
         return """Hey ${teamFlags.firstOrNull()?.owner?.name ?: "team"}!
-           |Launch Darkly thinks ${removableFlagCount(teamFlags)} of your ${teamFlags.size} flags could be ready for removal.
+           |LaunchDarkly thinks ${removableFlagCount(teamFlags)} of your ${teamFlags.size} flags could be ready for removal.
+           |The median age of the flags you maintain is ${teamFlags.getMedianAgeInDays()} days
            |Take a look ${flagReader.flagListUrl}""".trimMargin()
     }
 
     private fun removableFlagCount(flags: List<FlagDetail>): Int {
         return flags.filter { it.status == Status.REMOVABLE }.size
     }
+
+    companion object {
+
+        private const val MILLISECONDS_IN_A_DAY = 1000 * 60 * 60 * 24
+        fun List<FlagDetail>.getMedianAgeInDays(): Long = this.sortedBy { it.creationDate }.let { flagDetails ->
+            if (flagDetails.size % 2 == 0) {
```

Comment:
Excuse my lack of kotlin knowledge here. 

Is it possible to extract `flagDetails.size / 2` like statements to meaningful variables like `medianFlag` here to make this code easier to digest? 

Can we utilize a guard clause to reduce the cognitive complexity of this block?